### PR TITLE
Add example drawings between rounds

### PR DESCRIPTION
**Pull request by sondrefj**
## ❗️IMPORTANT❗️
For this feature to work, this PR: [Teknisk Museum Backend - PR](https://github.com/computas/tekniskmuseum-backend/pull/174) has to be pulled locally.

## Changes
_Added the section for singleplayer that shows example drawings to users in between rounds_
- Added a new service that is used to get drawings from backend
- The call for the images happens inside ngOnInit inside the game-intermediate-round component
- Used the new speech bubble component for feedback
- Added translations that are used to give the user feedback
- Restyled the intermediate screen to fit the screens on Teknisk Museum

## Notes
This only works on singleplayer atm. I have added a constraint that prevents this page from showing while in multiplayer. Translations are temporary. All changes from the splash component wasn't a part of this PR, but it got added to one of the commits after rebase.

## UI
Screenshot 2024-07-10 at 16 09 06

### CODE REVIEW

File: src/app/game/game-intermediate-result/game-example-drawings/game-example-drawings.component.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  ArrowAlignment = ArrowAlignment;
+  CustomColorsIO = CustomColorsIO;
+  constructor(
+    private translationService: TranslationService,
+    private drawingService: DrawingService,
+    private exampleDrawingService: ExampleDrawingService
+  ) {}
+  ngOnInit(): void {
+    this.label = this.drawingService.label;
+    this.exampleDrawingService.getExampleDrawings(3, this.label, this.language).subscribe((res) => {
```
Comment (ComputasAlex) [inline src/app/game/game-intermediate-result/game-example-drawings/game-example-drawings.component.ts]: Does this subscription get unsubscribed on component destruction?

File: src/app/game/game-intermediate-result/game-example-drawings/game-example-drawings.component.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+      this.exampleDrawings = res;
+    });
+  }
+
+  getMessage(): string {
+    const hasCorrectGuess = this.drawingService.lastResult.hasWon;
+    return this.getTranslatedMessage(this.label, this.language, hasCorrectGuess);
+  }
+
+  getTranslatedMessage(label: string, lang: SupportedLanguages, hasCorrectGuess: boolean): string {
```
Comment (ComputasAlex) [inline src/app/game/game-intermediate-result/game-example-drawings/game-example-drawings.component.ts]: Shouldn't translation be handled by translate pipe instead? For example like in game-drawing-display html component:
`<p class="round-score-text">+ {{ roundScore }} {{ 'POINTS' | translate }}</p>`

File: src/app/game/game-intermediate-result/game-example-drawings/game-example-drawings.component.html [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    class="speech-bubble right"
+    [pointerSide]="PointerSide.Top"
+    [arrowAlignment]="ArrowAlignment.Center"
+    [isFlipped]="false"
+    [textColor]="CustomColorsIO.black"
+    [bubbleColor]="CustomColorsIO.teal"
+    [bubbleSize]="'large'"
+  >
+    <div slot="body-alt" class="speech-bubble-text">
+      {{ getMessage() }}
```
Comment (ComputasAlex) [inline src/app/game/game-intermediate-result/game-example-drawings/game-example-drawings.component.html]: This should maybe use translation pipe as mentioned in the ts file?
E.g.: `<p class="round-score-text">+ {{ roundScore }} {{ 'POINTS' | translate }}</p>`

File: src/app/game/services/example-drawing.service.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+@Injectable({
+  providedIn: 'root',
+})
+export class ExampleDrawingService {
+  baseUrl = endpoints.TEKNISKBACKEND;
+
+  constructor(private http: HttpClient) {}
+
+  getExampleDrawings(numberOfDrawings: number, label: string, lang: SupportedLanguages): Observable<string[]> {
+    const body = '';
```
Comment (ComputasAlex) [inline src/app/game/services/example-drawing.service.ts]: Why do we send an empty string as the body?

File: src/assets/translation/EN.json [hunk 5]
```diff
@@ -59,5 +59,10 @@
   "ROUND_FEEDBACK_FAILED_DESC_2": "I can be a bit slow sometimes.. I'll do better next time!",
 
   "ROUND_FEEDBACK_CORRECT_TITLE": "Well drawn!",
-  "ROUND_FEEDBACK_CORRECT_DESC_1": "Thanks to you I guessed the correct word. Keep it up!"
+  "ROUND_FEEDBACK_CORRECT_DESC_1": "Thanks to you I guessed the correct word. Keep it up!",
+
+  "EXAMPLE_DRAWINGS_FB_CORRECT_P1": "I have learned that",
+  "EXAMPLE_DRAWINGS_FB_CORRECT_P2": "looks like this!",
```
Comment (ComputasAlex) [inline src/assets/translation/EN.json]: This is the same word as in FAILED_P2, could maybe use the same?

File: src/assets/translation/NO.json [hunk 6]
```diff
@@ -61,5 +61,10 @@
   "ROUND_FEEDBACK_FAILED_DESC_2": "Jeg kan være litt dum noen ganger..",
 
   "ROUND_FEEDBACK_CORRECT_TITLE": "Bra tegnet!",
-  "ROUND_FEEDBACK_CORRECT_DESC_1": "Takket være deg klarte jeg å gjette riktig."
+  "ROUND_FEEDBACK_CORRECT_DESC_1": "Takket være deg klarte jeg å gjette riktig.",
+
+  "EXAMPLE_DRAWINGS_FB_CORRECT_P1": "Jeg har lært at",
+  "EXAMPLE_DRAWINGS_FB_CORRECT_P2": "ser sånn ut!",
```
Comment (ComputasAlex) [inline src/assets/translation/NO.json]: This is the same word as in FAILED_P2, could maybe use the same?

**Review by ComputasAlex — CHANGES_REQUESTED** [review]
Amazing feature, and very nice implementation Sondre, great job! Can't wait to have this every time I play.

Made some suggestions about some minor details, please have a look. Also in backend PR

**Comment by ComputasAlex** [comment]
Also multiplayer screen looks a bit empty, but I guess we address this later

Skjermbilde 2024-07-11 kl  10 35 31

File: src/app/game/game-intermediate-result/game-example-drawings/game-example-drawings.component.ts (hunk 1 above)
Comment (sondrefj) [reply src/app/game/game-intermediate-result/game-example-drawings/game-example-drawings.component.ts]: Good point! It doesn't. Will fix 😎

File: src/app/game/services/example-drawing.service.ts (hunk 4 above)
Comment (sondrefj) [reply src/app/game/services/example-drawing.service.ts]: If I remember correctly, post requests made using the http-client from angular requires a body argument when made.

Comment (sondrefj) [reply src/app/game/services/example-drawing.service.ts]: I got an error when I tried to leave it out.

Comment (ComputasAlex) [reply src/app/game/services/example-drawing.service.ts]: Cleaned up with some more consistent structure here and in backend

**Review by ComputasAlex — APPROVED** [review]

**Pull request merged**